Documentation/planning/multi-sdk-band-support.md

```diff
@@ -0,0 +1,258 @@
+   2. **Checkout** Nxx branch
+   3. **Build** using `./build.sh --with-sdk <PSB Nxx SDK> --with-artifacts <Nxx inputs>`
+
+## Poison and Prebuilt detection
+
+Prebuilt detection should not change. Poisoning workflows may need to change. We don't want to poison inputs from the 1xx VMR, used when building the 2xx VMR. We **do** want to poison the previously source built inputs prior to adding in the 1xx inputs. So, in a poisoning check on a 2xx VMR build, the following workflow would be used:
+
+1. **Checkout** 1xx branch:
+2. **Build** using `./build.sh --with-sdk <PSB 1xx SDK> --with-artifacts <PSB 1xx artifacts>`.
+3. **Gather** 1xx build outputs from `artifacts/x64/Release` = `<CSB 1xx artifacts>`.
+4. **Prepare (pre-poison)** 2xx inputs - `<PSB 2xx artifacts> + <PSB 1xx artifacts>`
+5. **Poison** 2xx inputs
+6. **Augment (post-poison)** 2xx inputs - `<2xx inputs> <CSB 1xx artifacts>`
+7. **Checkout** 2xx branch
+8. **Build** using `./build.sh --with-sdk <PSB 2xx SDK> --with-artifacts <2xx inputs>`
+
+## Potential issues and mitigations
+
+There are a few potential issues that may require tweaks to this approach, source fixes, additional guidance, etc. They are listed below:
+
+### A newer SDK band may continue to require an older (typically 1xx band) SDK to build
+
+This could happen if a newer SDK band's (e.g. 2xx) components never take a dependency on newer 2xx features, yet do not successfully build with 2xx, due to new warnings, analyzers, etc. A two pronged approach is proposed to deal with this:
+- Microsoft documents the set of supported build combinations, including required inputs for all active SDK bands. This provides a guide for distro maintainers.
+- Where a band takes a dependency on a 'mix' of SDKs (e.g. dotnet/installer @ 2xx requires a 2xx SDK, but dotnet/roslyn cannot build with 2xx), Microsoft will fix input components so that the highest supported SDK band works across all the input components. So in this case, Microsoft would change dotnet/roslyn to build successfully with the 2xx SDK.
+
+### A toolset input to a newer band does not work with the PSB from that same band
+
+When preparing a newer band's input artifacts, there would be up to 3 versions of some components. For example:
+- PSB 1xx Microsoft.NET.Compilers.Toolset @ 4.7.1
+- CSB 1xx Microsoft.NET.Compilers.Toolset @ 4.7.2
+- PSB 2xx Microsoft.NET.Compilers.Toolset @ 4.8.0
+
+In this scenario, the *newest* version of Microsoft.NET.Compilers.Toolset will get used as input to any repo taking a dependency it it, before the 2xx dotnet/roslyn is built. If this particular repo has an explicit dependency on 4.7.x, then transparently updating to 4.8.x might cause a break. In the current source build infrastructure, the only way to avoid that transparent upgrade would be to not use the PSB 2xx inputs with the 2xx SDK build. But of course, the 2xx inputs would be required to provide supporting packages that the SDK depends on.
```

Comment:
```suggestion
In this scenario, the *newest* version of Microsoft.NET.Compilers.Toolset will get used as input to any repo taking a dependency on it, before the 2xx dotnet/roslyn is built. If this particular repo has an explicit dependency on 4.7.x, then transparently updating to 4.8.x might cause a break. In the current source build infrastructure, the only way to avoid that transparent upgrade would be to not use the PSB 2xx inputs with the 2xx SDK build. But of course, the 2xx inputs would be required to provide supporting packages that the SDK depends on.
```